packages/mnemonic-info-generator/src/agent/MnemonicInfoGenerator.ts

```diff
@@ -111,6 +113,46 @@ export class MnemonicInfoGenerator implements IAgentPlugin {
       .from(MnemonicInfo)
       .where('id = :id', { id: args.id })
       .orWhere('hash = :hash', { hash: args.hash })
-      .execute()
+      .execute();
+  }
+
+  private async saveMasterKey(args: IMnemonicInfoStoreArgs): Promise<UpdateResult> {
+    if (args.masterKey) {
+      return (await this.dbConnection).createQueryBuilder().update(MnemonicInfo).set({ masterKey: args.masterKey })
+          .where('id = :id', { id: args.id }).orWhere('hash = :hash', { hash: args.hash }).execute();
+    }
+    throw new Error('Master Key needs to be provided.');
+  }
+
+  private async generateMasterKey(args: IMnemonicInfoStoreArgs): Promise<IMnemonicInfoKeyResult> {
+    const mnemonic = (await this.getMnemonicInfo({ id: args.id, hash: args.hash })).mnemonic;
+    if (mnemonic) {
+      const seed = await bip39.mnemonicToSeed(mnemonic?.join(' '));
+      const { key, chainCode } = getMasterKeyFromSeed(seed.toString('hex'));
+      await this.saveMasterKey({ masterKey: key.toString('hex'), chainCode: chainCode.toString('hex') });
+      return { masterKey: key.toString('hex'), chainCode: chainCode.toString('hex') };
+    }
+    throw new Error('Mnemonic not found');
+  }
+
+  private async generateKeysFromMnemonic(args: IMnemonicInfoStoreArgs, context: IRequiredContext): Promise<ManagedKeyInfo> {
+    const mnemonic = (await this.getMnemonicInfo({ id: args.id, hash: args.hash })).mnemonic;
+    if (mnemonic && context) {
+      if (args.path && args.kms) {
+        const seed = await bip39.mnemonicToSeed(mnemonic?.join(' '));
+        const { key, chainCode } = derivePath(args.path, seed.toString('hex'), args.offset);
```

Comment:
Fixed!

Reply:
Where?

Reply:
It's not ready, the ticket still assigned to me. ;-)

Reply:
Okay then I suggest to update your flow, to inform in PRs after you have submitted it, because that means async work between team members is not really possible